pkg/gitrepo/repo.go

```diff
@@ -404,6 +404,11 @@ func (r *Repo) checkoutRef(ref string) (*git.Worktree, error) {
 		return nil, microerror.Mask(err)
 	}
 
+	err = worktree.Clean(&git.CleanOptions{Dir: true})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
How did you even discover this is a problem?

I'm thinking whether we shouldn't switch to calling `git` binary.

Reply:
I was debugging a totally unrelated opsctl problem that made it pretty transparent that we're leaking directories when switching branches. Using the git binary will make things a lot more predictable I think, my only concern is that I don't know if we can use a token to authenticate in that case.

Reply:
> my only concern is that I don't know if we can use a token to authenticate in that case

I don't know off top of my head.

Reply:
Looks like you can https://github.blog/2012-09-21-easier-builds-and-deployments-using-git-over-https-and-oauth/